src/hb-ot-font.cc

```diff
@@ -228,6 +228,32 @@ hb_ot_get_glyph_from_name (hb_font_t *font HB_UNUSED,
 }
 #endif
 
+#define _HB_OT_STYLE_SLANT HB_TAG ('s','l','n','t')
+#define _HB_OT_STYLE_ITALIC HB_TAG ('i','t','a','l')
+
+static float
+hb_ot_get_font_italic_angle (hb_font_t *font)
+{
+#ifndef NO_VAR
+  hb_ot_var_axis_info_t info;
+  /* Check first if is set by variation as
+     https://docs.microsoft.com/en-us/typography/opentype/spec/dvaraxistag_slnt */
+  if (hb_ot_var_find_axis_info (font->face, _HB_OT_STYLE_SLANT, &info) &&
+      info.axis_index < font->num_coords)
+    return (float) font->coords[info.axis_index] / 16384.f;
+
+  if (hb_ot_var_find_axis_info (font->face, _HB_OT_STYLE_ITALIC, &info) &&
+      info.axis_index < font->num_coords)
+    return (float) font->coords[info.axis_index] * -12.f / 16384.f;
```

Comment:
Italic axis is in 0..1 range and 12° apparently is what italic means in Arial, and it should be negative for "forward" (!) values, https://docs.microsoft.com/en-us/typography/opentype/spec/post

Reply:
I don't think we should be trying to convert an 'ital' axis value to an italic angle at all.

Perhaps there should be an MVAR tag to allow the 'post' table's italicAngle field to respond to variations. In the absence of that, the most reasonable heuristic to synthesize an italicAngle for a variation font, if we have some reason to believe it *should* vary, would be based on the caretSlopeRise and caretSlopRun values.

Reply:
Actually.... my bad again.  I meant `slnt` axis, not `ital`.  In fact, there's enough words in the spec to tell us exactly what to do.  From MVAR spec:
```
Note: The OS/2.usWeightClass, OS/2.usWidthClass and post.italicAngle values are not supported by variation data in the 'MVAR' table. This is because values for these three fields correspond directly to input axis values for the 'wght', 'wdth' and 'slnt' variation axes. See the discussion of these axes in the OpenType Design-Variation Axis Tag Registry for details on the relationship between these fields and the corresponding design axes.
```
https://docs.microsoft.com/en-us/typography/opentype/otspec182/mvar

From the axis tags description for `slnt`:
```
Note that the scale for the Slant axis is interpreted as the angle of slant in counter-clockwise degrees from upright. This means that a typical, right-leaning oblique design will have a negative slant value. This matches the scale used for the italicAngle field in the 'post' table.

In a variable font that implements 'slnt' variations, the value in the italicAngle field of the 'post' table must match the default 'slnt' value specified in the 'fvar' table. For non-default instances of a variable font, the 'slnt' axis value can be used as the post.italicAngle value for the instance.
```
https://docs.microsoft.com/en-us/typography/opentype/otspec182/dvaraxistag_slnt

The spec for `wght` and `wdth` also have similar wording in reference to their OS/2 counterparts.

So!  I propose we add the three tags `slnt`, `wght`, and `wdth` to access those three values in the ot-metrics API.  These will return the currently-set variation value for the item (in user-space, which we currently don't keep but we want to) definitely clamped to the valid range only, and falling back to the `post` / `OS/2` table values.

If we add separate ot-style API later, we probably will expose the same logic there too...  Or maybe not.  That API may only expose what's encoded in `STAT` only, with very limited fallback if there's no `STAT` at all, in that case maybe exposing the three items.